android/ScratchJr/app/src/main/java/org/scratchjr/android/ScratchJrUtil.java

```
@@ -190,6 +190,13 @@ private static String getLastPathComponent(String filePath) {
             while ((ze = zin.getNextEntry()) != null) {
                 String path = toPath + File.separator + ze.getName();
                 File unzipFile = new File(path);
+                // Zip files can contain an entry (file or directory) having path
+                // traversal characters ("../") in its name. Before unzipping,
+                // we need to confirm it will only extract to the expected folder.
+                // For more details see https://support.google.com/faqs/answer/9294009
+                if (!unzipFile.getCanonicalPath().startsWith(toPath)) {
+                    continue;
```

We should be throwing `SecurityException` here. Maybe instead of `continue;` we have `throw SecurityException("Unsafe file path found and unzipping will not be allowed for security purposes.");`